api/types/saml_idp_service_provider.go

```diff
@@ -21,15 +21,25 @@ import (
 	"fmt"
 
 	"github.com/gravitational/trace"
+	"golang.org/x/exp/slices"
 
 	"github.com/gravitational/teleport/api/utils"
 )
 
+const (
+	unspecifiedNameFormat = "urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified"
+	uriNameFormat         = "urn:oasis:names:tc:SAML:2.0:attrname-format:uri"
+	basicNameFormat       = "urn:oasis:names:tc:SAML:2.0:attrname-format:basic"
+)
+
 var (
 	// ErrMissingEntityDescriptorAndEntityID is returned when both entity descriptor and entity ID is empty.
-	ErrEmptyEntityDescriptorAndEntityID = trace.BadParameter("either entity_descriptor or entity_id must be provided")
+	ErrEmptyEntityDescriptorAndEntityID = &trace.BadParameterError{Message: "either entity_descriptor or entity_id must be provided"}
```

Comment:
Why did you change this, and errors below, from using `trace.BadParameter()` function?

Reply:
I received a suggestion in my backport that using `trace.BadParameter` directly will reference the location where it is defined instead of the line where the actual error occurs. But using `trace.BadParameterError` struct resolves that and prints a proper stack trace.